code/modules/mod/modules/modules_engineering.dm

```diff
@@ -260,3 +260,75 @@
 #undef EXTINGUISHER
 #undef NANOFROST
 #undef METAL_FOAM
+
+/obj/item/mod/module/drone
+	name = "MOD Personal Artificial Intelligence Storage"
+	desc = "A module that can be used to store a personal AI device in. \
+			Will convert the personal AI into a construction drone to help \
+			the owner with any repair work"
+	icon_state = "tether"
+	module_type = MODULE_USABLE
+	complexity = 1
+	use_power_cost = DEFAULT_CHARGE_DRAIN
+	incompatible_modules = list(/obj/item/mod/module/drone)
+	cooldown_time = 3 SECONDS
+	allow_flags = MODULE_ALLOW_INACTIVE
+	/// The drone that we currently have stored
+	var/mob/living/silicon/robot/drone/stored_drone
+	/// The drone that is linked to the mod
+	var/mob/living/silicon/robot/drone/linked_drone
+	/// The mob that summoned the drone. Used to clear the trait
+	var/mob/living/carbon/human/summoner
```

Comment:
You should have a qdel signal listening to this, to properly clean up this ref, or just use a UID here

Reply:
A signal sounds fair, I'll implement it